integration/use-cases/lib/utils.js

```diff
@@ -1,11 +1,25 @@
+const {
+  screenshotsFolder,
+} = require("../../args");
+const path = require("path");
+
 module.exports = {
-  retryAndRefresh: async (page, retries, toCheck) => {
+  retryAndRefresh: async (page, retries, toCheck,  testName) => {
     let retriesLeft = retries;
     while (retriesLeft > 0) {
       try {
         await toCheck();
         break;
       } catch (e) {
+        if(testName) {
+          await page.screenshot({
+            path: path.join(__dirname, `../../${screenshotsFolder}/${testName}-${retries - retriesLeft}.png`),
+          });
+        }
```

Comment:
this is mostly for debugging a specific step with several retries. I have removed the places in which I used it (to avoid polluting results with other screenshots) but left this code in case we need it in the future.